src/specs/networking.md

```diff
@@ -66,3 +71,40 @@ Accepting a `WireTxPayForMessage` into the mempool requires different logic than
+| `shareProofs` | [ShareProof](#shareproof)`[]`                                                         | The available shares in the offending row or column.                                                 |
+| `isCol`       | `bool`                                                                                | A Boolean indicating if it is an offending row or column; `false` if it is a row.                     |
+| `position`    | `uint64`                                                                              | The index of the offending row or column in the square.                                               |
+## Invalid State Update
+
+If a malicious block producer incorrectly computes the state, a fraud proof for this can be presented. We assume that the light clients have the [AvailableDataHeader](./data_structures.md#availabledataheader) and the [Header](./data_structures.md#header) for each block. Hence, given a [ShareProof](#shareproof), they can verify if the `rowRoot` or `colRoot` specified by `isCol` and `position` commits to the corresponding [Share](./data_structures.md#share). Similarly, given the `height` of a block, they can access all elements within the [AvailableDataHeader](./data_structures.md#availabledataheader) and the [Header](./data_structures.md#header) of the block. 
+
+### StateFraudProof
+
+| name                       | type                                                                                     | description                                                                                                                                                                                            |
+|----------------------------|------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `height`                   | [Height](./data_structures.md#type-aliases)                                              | Height of the block with the intermediate state roots. Subtracting one from `height` gives the height of the block with the transactions.                                                             |
+| `transactionShareProofs`   | [ShareProof](#shareproof)`[]`                                                            | `isCol` of type `bool` must be `false`.                                                                                                                                                                |
+| `stateShareProofs`         | [ShareProof](#shareproof)`[]`                                                            | `isCol` of type `bool` must be `false`.                                                                                                                                                                |
```

Comment:
What's a state share proof? State isn't in arranges in shares afaik.

Reply:
stateShareProofs is a list of 'ShareProof's for the 'Share's that contain the intermediate state roots. The name seems to be misleading so I changed it to isrShareProofs.